<?xml version="1.0"?>

<robot name="rod2d"
    xmlns="http://drake.mit.edu"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xsi:schemaLocation="http://drake.mit.edu ../../doc/drakeURDF.xsd">

  <link name="base">
    <visual>
      <origin xyz="0 0 -0.50" rpy="0 0 0"/>
      <geometry>
        <box size="3 3 1"/>
      </geometry>
      <material name="green">
        <color rgba="0 1 0 1" />
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 -0.50" rpy="0 0 0"/>
      <geometry>
        <box size="3 3 1"/>
      </geometry>
    </collision>
  </link>   


  <link name="pusher">
    <inertial>
      <origin xyz="0 -0.01 0" rpy="0 0 0" />
      <mass value="1" />
      <inertia ixx="1" ixy="0" ixz="0" iyy="1" iyz="0" izz="1" />
    </inertial>
    <visual>
      <origin xyz="0 -0.01 0" rpy="0 0 0"/>
      <geometry>
        <box size="1 0.02 0.1"/>
      </geometry>
    </visual>
    <collision>
      <origin xyz="0 -0.01 0" rpy="0 0 0"/>
      <geometry>
        <box size="1 0.02 0.1"/>
      </geometry>
    </collision>
  </link> 

 	<link name="rod">
 		<inertial>
      <origin xyz="0 0 0" rpy="0 1.5708 0"/>
      <mass value="1" />
      <inertia ixx="1" ixy="0" ixz="0" iyy="0.083" iyz="0" izz="1" />
    </inertial>

 		<visual>
 			<origin xyz="0 0 0" rpy="0 1.5708 0"/>
 			<geometry>
 				<cylinder length="0.5" radius="0.005"/>
 			</geometry>
 		</visual>
    <visual>
      <origin xyz="0.25 0 0"/>
      <geometry>
        <sphere radius="0.01" />
      </geometry>
    </visual>
    <visual>
      <origin xyz="-0.25 0 0"/>
      <geometry>
        <sphere radius="0.01" />
      </geometry>
    </visual>

    <collision>
      <origin xyz="0.25 0 0"/>
      <geometry>
        <sphere radius="0.01" />
      </geometry>
    </collision>
    <collision>
      <origin xyz="-0.25 0 0"/>
      <geometry>
        <sphere radius="0.01" />
      </geometry>
    </collision>
 	</link>	

  <joint name="pusher_base" type="prismatic">
    <parent link="base"/>
    <child link="pusher"/>
    <origin xyz="0 0 0" rpy="0 0 0"/>
    <axis xyz="0 1 0"/>
  </joint>

  <joint name="rod_base" type="floating">
    <parent link="base"/>
    <child link="rod"/>
  </joint>

 </robot>